internal/models/api/token.go

```diff
@@ -0,0 +1,52 @@
+// Package internal/models/api defines models to be used for API requests.
+package api
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/SimplQ/simplQ-golang/internal/models/db"
+)
+
+// AddTokenRequest is a model to structure an add token request.
+//
+// Members
+//
+// QueueId          -   QueueId of the queue that this token is to be added to
+// Name             -   Name of the token, typically name of the person whom the
+//                      token was issued to.
+// ContactNumber    -   Contact Number.
+// EmailId          -   Optional. Email ID if the queue collects email ID of users.
+type AddTokenRequest struct {
+	QueueId       db.QueueId
+	Name          string
+	ContactNumber string
+	EmailId       string
+}
+
+// AddTokenResponse is a model to structure the response of an add token request.
+type AddTokenResponse db.TokenId
+
+func (req AddTokenRequest) Validate() (ValidationError, bool) {
+	if len(req.Name) < MIN_LENGTH || len(req.Name) > MAX_LENGTH {
```

Comment:
Where is MIN_LENGTH and MAX_LENGTH defined?

Reply:
It is actually defined in `internal/models/api/queue.go`
So, its using the same constraints as `QueueName` but we can change it to `MIN_LENGTH_QUEUE_NAME` and `MIN_LENGTH_TOKEN_NAME` etc.

Reply:
Changing and moving to right files would be a good idea. I don't want to in future set queue name to be shorter, and inadvertently limit token name length.

Reply:
Changed constant names and put them in correct files.